Dynatrace/dynatrace-operator
# Add k8s app enabled feature flag

## Description

To automatically enable Kubernetes cluster for the Kubernetes 3gen app during Dynatrace deployment process with (kubectl apply -f dynakube.yaml) newly provided feature flag on operator (dynakube.yaml) can be used: **feature.dynatrace.com/k8s-app-enabled**

In case FF is set to true (feature.dynatrace.com/k8s-app-enabled: "true"), then setting object is created accordingly ("schemaId": "builtin:app-transition.kubernetes").

Creation of schema object ("schemaId": "builtin:app-transition.kubernetes") is done with scope of newly created cluster ME, which was created with creation of schema object with "schemaId": "builtin:cloud.kubernetes".

**Expected behavior for feature flags in the operator is:**
FF true --> explicitly set enableApp to true
FF false --> don't do anything
FF not there --> don't do anything.

## How can this be tested?

The result can be validated on cluster settings page 'Kuberneted App' on new cluster creation, toggle 'Enable Kubernetes App' should be enabled
![image](https://github.com/Dynatrace/dynatrace-operator/assets/138136764/9d62c975-a10e-496a-9d07-4aa4249a3f78)

*Important*
settings schema version: it will be 1.0.0 with future release of Kubernetes App settings on cluster, but you should be able to use the current version 0.1.0, as regarding properties, there should be no difference between the versions.

## Checklist
